packages/langium/src/lsp/language-server.ts

```diff
@@ -117,6 +118,9 @@ export class DefaultLanguageServer implements LanguageServer {
                 implementationProvider: hasGoToImplementationProvider,
                 callHierarchyProvider: hasCallHierarchyProvider
                     ? {}
+                    : undefined,
+                documentLinkProvider: documentLinkProvider
+                    ? { resolveProvider: !!documentLinkProvider.resolveDocumentLink }
                     : undefined
```

Comment:
This would also change if above is accepted
```suggestion
                documentLinkProvider: hasDocumentLinkProvider
                    ? { resolveProvider: !!this.services.lsp.DocumentLinkProvider?.resolveDocumentLink }
                    : undefined
```